__tests__/apps.test.js

```diff
@@ -0,0 +1,99 @@
+const request = require('supertest');
+const app = require('../app.js');
+const connect = require('../db/connection.js');
+const data= require('../db/data/test-data');
+const seed = require('../db/seeds/seed.js');
+const endpoints= require('../endpoints.json');
+
+beforeEach(()=> seed(data))
+afterAll(() => connect.end());
+
+
+describe('all bad urls', () => {
+    it('should respond with 404 and message not found!', ()=>{
+        return request(app)
+        .get('/api/sksafd')
+        .expect(404)
+        .then((res)=>{
+            expect(res.body.msg).toBe(`not found!`)
+            
+        })
+    })
+})
+
+describe('all bad urls', () => {
+    it('should respond with 404 and message not found!', ()=>{
+        return request(app)
+        .get('/api/sksafd')
+        .expect(404)
+        .then((res)=>{
+            expect(res.body.msg).toBe(`not found!`)
+            
+        })
+    })
+});
+
+describe('GET /api/topics', () => {
+    it('should respond with an array of all topics', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then((res) => {
+            expect(res.body.topics.length).toBe(3);
+            res.body.topics.forEach((topic)=> {
+                expect(topic).toMatchObject({
+                    description: expect.any(String),
+                    slug: expect.any(String)
+                })
+            })
+        });
+    });
+    
+});
+
+describe('GET /api', () => {
+    it('respond with JSON containing all endpoints information', () => {
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then((res) => {
+            Object.values(res.body).forEach(endpoint => {
```

Comment:
Here we can just assert that the return value is equal to the json (which we can require in)